client/src/routes/login.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+	import Textfield, { HelperLine } from '@smui/textfield';
+	import Button, { Label as ButtonLabel } from '@smui/button';
+	import Snackbar, { Label as SnackbarLabel, SnackbarComponentDev } from '@smui/snackbar';
+	import { form, field } from 'svelte-forms';
+	import { required } from 'svelte-forms/validators';
+	import { user_api } from '../api/user';
+
+	let failureSnackbar: SnackbarComponentDev;
+
+	const user = field('user', '', [required()]);
+	const password = field('password', '', [required()]);
+	const loginForm = form(user, password);
+
+	function login() {
+		if ($loginForm.valid && $user.value && $password.value) {
+			user_api
+				.login({ user: $user.value, password: $password.value })
+				.then((res) => {
+					// TODO something
+					alert('Success - ' + JSON.stringify(res.headers));
+				})
+				.catch((_err) => {
+					alert('Error');
+				});
+		} else {
+			failureSnackbar.close();
+			failureSnackbar.open();
+		}
+	}
+</script>
+
+<div class="form-container">
+	<Textfield
+		variant="outlined"
+		bind:value={$user.value}
+		label="Username"
+		bind:dirty={$user.dirty}
+		bind:invalid={$user.invalid}
+		><svelte:fragment slot="helper">
+			{#if !$user.valid}
+				<HelperLine>Not a valid username</HelperLine>
+			{/if}
+		</svelte:fragment>
+	</Textfield>
+
+	<Textfield
+		type="password"
+		variant="outlined"
+		bind:value={$password.value}
+		label="Password"
+		bind:dirty={$password.dirty}
+		bind:invalid={$password.invalid}
+		><svelte:fragment slot="helper">
+			{#if !$password.valid}
+				<HelperLine>Not a valid password</HelperLine>
+			{/if}
+		</svelte:fragment>
+	</Textfield>
+
+	<Button type="submit" on:click={login} variant="raised">
+		<ButtonLabel>Login</ButtonLabel>
+	</Button>
+</div>
+
+<Snackbar bind:this={failureSnackbar}>
+	<SnackbarLabel>Please specify valid credentials!</SnackbarLabel>
+</Snackbar>
+
+<style lang="scss">
+	.form-container {
+		display: flex;
+		flex-direction: column;
+		align-items: center;
+	}
+
+	* :global(.mdc-text-field) {
```

Comment:
These are scoped, right? I assume svelte takes care of it.
This scss :global looks serious 😀

Reply:
Without it the css changes do not register, but I will look into it, svelte really should take care of it otherwise it would be really inconvenient to style the material ui.

Reply:
It **IS** scoped thankfully, I tried it with `background: red;` and it only changed things in one component.